File: Code/Level-1/todo-fullstack/todo-web-ui/src/Todo.js
//---------------------------------------
//Model
//---------------------------------------

class Todo {
  static nextId = 0;
  constructor(title) {
    this.id = ++Todo.nextId;
    this.title = title;
    this.completed = false;
  }
}

// module.exports = Todo; // cjm
// or
export default Todo; // esm

File: Code/Level-3/todo-service/src/main/java/com/example/model/Todo.java
package com.example.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;


@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name="todos")
public class Todo {

    @Id
    @GeneratedValue
    private int id;
    private String title;
    private boolean completed;

    @ManyToOne
    @JoinColumn(name="user_id")
    private User user;

}
